helix-term/src/lib.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::type_complexity)]
```

Comment:
clippy generated a bunch of warnings here which would require introducing new types.
I held off doing that to avoid introducing too many changes here.

Reply:
I think it would be ok to resolve those in this PR as well as long as it's a separate commit from the existing changes